main/minimize/minimize.cc

```diff
@@ -0,0 +1,351 @@
+#include "main/minimize/minimize.h"
+#include "absl/strings/match.h"
+#include <optional>
+
+using namespace std;
+
+namespace sorbet::realmain {
+
+namespace {
+
+const vector<core::SymbolRef> IGNORED_BY_SYMBOL = {
+    // Kernel is very common and produces a lot of missing_methods churn, but is rarely necessary
+    // for type-checking. (It also sometimes gets inferred even when Sorbet already statically knows
+    // about Kernel, which can cause dangling references to deleted classes.)
+    core::Symbols::Kernel(),
+};
+
+const vector<string_view> IGNORED_BY_FULL_NAME = {
+    // These modules are Sorbet-internal and aren't necessary for static typechecking, so they
+    // aren't necessary and they cause a fair bit of missing_methods churn
+    "T::Private::Methods::MethodHooks",    "T::Private::Methods::SingletonMethodHooks",
+    "T::Private::Abstract::Hooks",         "T::Private::MixesInClassMethods",
+    "T::Private::Final::NoInherit",        "T::Private::Final::NoIncludeExtend",
+    "T::Private::Sealed::NoIncludeExtend", "T::InterfaceWrapper::Helpers",
+};
+
+enum class OutputCategory {
+    External = 1,
+    Util,
+    Opus,
+    GraphQL,
+    AutogenProto,
+    Autogen,
+    Flatfiles,
+    Model,
+    Mutator,
+    Chalk,
+};
+
+OutputCategory outputCategoryFromClassName(string_view fullName) {
+    if (absl::StrContains(fullName, "Flatfiles")) {
+        return OutputCategory::Flatfiles;
+    } else if (absl::StrContains(fullName, "::Mutator")) {
+        return OutputCategory::Mutator;
+    } else if (absl::StrContains(fullName, "::Model")) {
+        return OutputCategory::Model;
+    } else if (absl::StrContains(fullName, "::Model")) {
+        if (absl::StartsWith(fullName, "Plaid")) {
+            return OutputCategory::External;
+        } else {
+            return OutputCategory::Model;
+        }
+    } else if (absl::StrContains(fullName, "::Autogen::Proto")) {
+        return OutputCategory::AutogenProto;
+    } else if (absl::StrContains(fullName, "::Autogen")) {
+        return OutputCategory::Autogen;
+    } else if (absl::StrContains(fullName, "::GraphQL")) {
+        return OutputCategory::GraphQL;
+    } else if (absl::StartsWith(fullName, "Opus")) {
+        return OutputCategory::Opus;
+    } else if (absl::StartsWith(fullName, "Chalk")) {
+        return OutputCategory::Chalk;
+    } else if (absl::StartsWith(fullName, "PrisonGuard") || absl::StartsWith(fullName, "Critic") ||
+               absl::StartsWith(fullName, "Primus")) {
+        return OutputCategory::Util;
+    } else {
+        return OutputCategory::External;
+    }
+}
+
+core::MethodRef closestOverridenMethod(const core::GlobalState &gs, core::SymbolRef enclosingClassSymbol,
```

Comment:
```suggestion
core::MethodRef closestOverridenMethod(const core::GlobalState &gs, core::ClassOrModuleRef enclosingClassSymbol,
```

Reply:
i'll cling to my SymbolRef's kicking and screaming until it's the only option 😅